test/middleware-config/pages/middlewareTelefunc.ts

```diff
@@ -28,10 +30,7 @@ const telefuncUniversalMiddleware: UniversalMiddleware = async (request, context
   })
 }
 
-const middlewareTelefunc: Middleware[] = [
-  {
-    name: 'telefunc',
-    order: 'pre',
-    value: telefuncUniversalMiddleware
-  }
-]
+const middlewareTelefunc = decorate(telefuncUniversalMiddleware, {
+  method: 'POST',
+  path: '/_telefunc'
+})
```

Comment:
`withRoute` replaced by a more generic `decorate` helper, to make things more explicit, and simplify adding props in the future

Reply:
Good idea to keep it generic. (Name idea: `enhance()` instead of `decorate()`, but I'm fine with `decorate()` if you prefer it.)

Reply:
`enhance` was my first name idea, but I preferred `decorate`. I'll let that macerate a little and see how I feel about it when we're closer to a finished implementation.

Reply:
:+1: The thing I prefer about `enhance()` is that it's more visual and concrete, whereas `decorate()` is more abstract and vague (or maybe it's subjective), thus more difficult to remember.

Bonus: in the docs we can mention "enhanced middlewares", for example:
 - > Enhanced middlewares are middlewares that define a route
 - > Enhanced middlewares must have a name

Reply:
But I guess we could also say "decorated middlewares".

Reply:
In the end I want for `enhance` :)